From 71771621b4de78c480c8e4731f8fadcdbf5993c7 Mon Sep 17 00:00:00 2001
From: Varun Sharma <vsharma@habana.ai>
Date: Thu, 21 Apr 2022 06:54:51 +0300
Subject: [PATCH] [SW-84007] greco: pci: enable pci axi drain

Enabling both hbw and lbw drain in the pci block, so if the NOC is
stuck the pci transaction will continue to flow and the pci block
will continue to function.

Change-Id: Iac94b3807486978ddfbfa7bd776fc44987e31a3f
Signed-off-by: Varun Sharma <vsharma@habana.ai>
---

diff --git a/arch/greco/common/common.c b/arch/greco/common/common.c
index f1fbf99..9d895be 100644
--- a/arch/greco/common/common.c
+++ b/arch/greco/common/common.c
@@ -838,3 +838,15 @@
 	greco_ack_pci_axi_irq();
 	greco_ack_gic_irq();
 }
+
+#define HBW_DRAIN_TIMEOUT      0x1000
+#define LBW_DRAIN_TIMEOUT      0x5000
+
+void ppboot_pcie_drain_init(void)
+{
+	regmap_writel(HBW_DRAIN_TIMEOUT, mmPCIE_WRAP_HBW_DRAIN_TIMEOUT);
+	regmap_writel(1, mmPCIE_WRAP_HBW_DRAIN_CFG);
+
+	regmap_writel(LBW_DRAIN_TIMEOUT, mmPCIE_WRAP_LBW_DRAIN_TIMEOUT);
+	regmap_writel(1, mmPCIE_WRAP_LBW_DRAIN_CFG);
+}
